### DISCUSSION TITLE
Error message: The study model does not support required optimizer

### DISCUSSION THREAD

**Discussion by bionut00** [edwardkort/WWIDesigner#157] [NAF Study] [answered]
Hello,
I am trying to learn the WWIDesigner, i followed the video tutorial but when i want to open the contraints file i get the following message: the study model does not support required optimizer, HoleFromTopObjectiveFUnction (Hole size & Position)
How can o get around that?

**Comment by edwardkort** [comment]
It sounds like you are trying to load an older version of a constraints file, or you hand edited the constraints file. Just use (and possibly modify) a constraints file that was distributed with the version of WIDesigner that you are using.

**Comment by bhp1** [comment, marked as answer]
Which study model (type of instrument) are you using?  The HoleFromTopObjectiveFunction optimizer works only with the NAF study model, not with whistles, flutes, or reeds.  Use the Options dialog to set a [study model](https://github.com/edwardkort/WWIDesigner/wiki/WIDesigner-Options#study-model).  To start out, pick an optimizer from the list in the instrument study panel on the left hand side rather than opening a constraints file.  If you do pick a constraints file, be sure to use one intended for the study model you have set in the options,

**Reply by bionut00** [reply]
Thanks! You are right, i had it set on whistle but i was trying to make a NAF.
